### PR TITLE
Use Chef's built-in apt resource

### PR DESCRIPTION
**Pull request by eherot**
Tell me whether or not you would consider this a breaking change (esp. considering the requirement of using a specific Chef version).

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -11,5 +11,6 @@
 
 supports 'ubuntu', '= 14.04'
 
+chef_version '~> 12.10'
```
Comment (jeffbyrnes) [inline metadata.rb]: I was unaware of this; awesome.

File: recipes/default.rb [hunk 2]
```diff
@@ -17,7 +17,14 @@
...
 #
 
-include_recipe 'apt' if platform_family?('debian')
+if Chef::VersionConstraint.new('< 12.10.0').include? Chef::VERSION
+  raise 'This recipe requires Chef version 12.10 or greater'
+end
+
+apt_update 'et_fog' do
+  action :nothing
+end.run_action(:periodic)
```
Comment (jeffbyrnes) [inline recipes/default.rb]: This is, I suspect, to ensure it’s run at compile time?

**Comment by jeffbyrnes** [comment]
Hmm. I suppose this is a breaking change, yeah, esp. in light of this being a community cookbook.

**Comment by jeffbyrnes** [comment]
Looks good though.

File: recipes/default.rb (hunk 2 above)
Comment (eherot) [inline recipes/default.rb]: Yes. Mandatory because the rest of the recipe basically gets processed at compile time.

**Pull request merged**